### ISSUE TITLE
Runtime exception was caught 'Unknown datatype passed to getBool' for decimal preprocessor comparison

### ISSUE THREAD

**Issue by cverbiest** [Riverside-Software/sonar-openedge#235]
sonarlint fails on following code
```abl
&global-define app_version 16.10

define variable UnusedVar as character no-undo.

&if {&app_version} <= 12.10
&then
    &Scoped-define with_catch no
&else
    &Scoped-define with_catch yes
&endif
```

Error and stack
```
Parsing C:/Users/cvb/git/ccesmarttools/CceSmartTools/sonar_issue_main.p
Runtime exception was caught 'Unknown datatype passed to getBool' - Please report this issue : 
  org.sonar.plugins.openedge.api.org.prorefactor.proparse.antlr4.PreproEval.getBool(PreproEval.java:521)
  org.sonar.plugins.openedge.api.org.prorefactor.proparse.antlr4.PreproEval.visitPreproIfEval(PreproEval.java:77)
  org.sonar.plugins.openedge.api.org.prorefactor.proparse.antlr4.PostLexer.preproIfCond(PostLexer.java:261)
  org.sonar.plugins.openedge.api.org.prorefactor.proparse.antlr4.PostLexer.preproIf(PostLexer.java:154)
  org.sonar.plugins.openedge.api.org.prorefactor.proparse.antlr4.PostLexer.nextToken(PostLexer.java:61)
  org.sonar.plugins.openedge.api.org.prorefactor.proparse.antlr4.PostLexer.nextToken(PostLexer.java:34)
  org.sonar.plugins.openedge.api.org.prorefactor.proparse.antlr4.TokenList.build(TokenList.java:56)
  org.sonar.plugins.openedge.api.org.prorefactor.proparse.antlr4.TokenList.nextToken(TokenList.java:152)
  org.sonar.plugins.openedge.api.org.prorefactor.proparse.antlr4.MultiChannelTokenSource.nextToken(MultiChannelTokenSource.java:43)
  org.sonar.plugins.openedge.api.org.prorefactor.proparse.antlr4.ProgressLexer$ANTRL2TokenStreamWrapper.nextToken(ProgressLexer.java:1001)
  org.sonar.plugins.openedge.api.org.prorefactor.treeparser.ParseUnit.lex(ParseUnit.java:123)
  org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.execute(OpenEdgeProparseSensor.java:108)
  org.sonarsource.sonarlint.core.analyzer.sensor.SensorWrapper.analyse(SensorWrapper.java:52)
  org.sonarsource.sonarlint.core.analyzer.sensor.SensorsExecutor.executeSensor(SensorsExecutor.java:75)
  org.sonarsource.sonarlint.core.analyzer.sensor.SensorsExecutor.execute(SensorsExecutor.java:63)
  org.sonarsource.sonarlint.core.analyzer.sensor.PhaseExecutor.execute(PhaseExecutor.java:44)
  org.sonarsource.sonarlint.core.container.analysis.AnalysisContainer.doAfterStart(AnalysisContainer.java:143)
  org.sonarsource.sonarlint.core.container.ComponentContainer.startComponents(ComponentContainer.java:125)
  org.sonarsource.sonarlint.core.container.ComponentContainer.execute(ComponentContainer.java:110)
  org.sonarsource.sonarlint.core.container.storage.StorageAnalyzer.analyze(StorageAnalyzer.java:72)
  org.sonarsource.sonarlint.core.container.storage.StorageContainer.analyze(StorageContainer.java:126)
  org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.analyze(ConnectedSonarLintEngineImpl.java:158)
  org.sonarsource.sonarlint.core.ConnectedSonarLintEngineImpl.analyze(ConnectedSonarLintEngineImpl.java:145)
  org.sonarlint.eclipse.core.internal.server.Server.runAnalysis(Server.java:273)
  org.sonarlint.eclipse.core.internal.jobs.AnalyzeProjectJob.run(AnalyzeProjectJob.java:469)
  org.sonarlint.eclipse.core.internal.jobs.AnalyzeProjectJob$AnalysisThread.run(AnalyzeProjectJob.java:128)
1 files proparse'd, 0 failure(s)
```

**Comment by gquerret** [comment]
Test case can even be reduced to:
```
&if 1 <= 2 &then &endif
```

**Comment by gquerret** [comment]
Wow...
https://github.com/Riverside-Software/sonar-openedge/commit/38d80375cded806f705e0744706882572ce0cb01
